API_SPEC.md

```diff
@@ -0,0 +1,305 @@
+1. [Submissions](#submissions)
+    1. [GET /submissions/:id](#get-submissionsid)
+    2. [PUT /submissions/:id](#put-submissionsid)
+    3. [DELETE /submissions/:id](#delete-submissionsid)
+2. [Comments](#comments)
+    1. [GET /comments/:id](#get-commentsid)
+    2. [POST /submissions/:submission_id/comments/:id](#post-submissionssubmission_idcommentsid)
+    3. [PATCH /comments/:id](#patch-commentsid)
+    4. [POST /comments/:id/preview](#post-commentsidpreview)
+
+## Submissions
+### GET /submissions/:id
+Gets submission object
+
+#### Responses
+**200**:
+```
+{
+    "_link": "/submissions/d5f76a20f4b847c59d2d9b65f134c9ef",
+    "views": "4",
+    "solution": "Solution code here...",
+    "acceptsFeedback": "true",
```

Comment:
The current behavior is that people can always post feedback, no matter what state of the exercise is ("pending", "hibernating", "done").